firebase-project/src/views/login.vue

```diff
@@ -0,0 +1,145 @@
+      LOGIN FORM
+    </h1>
+    <Form class="w-3/4 mx-auto" @submit="handleLogin">
+      <label class="flex text-white mt-3 mb-1 text-lg">Email</label>
+      <Field
+        class="w-full border-solid outline-none  tracking-wider p-2 text-sm md:text-base lg:text-lg"
+        v-model="user.email"
+        name="email"
+        type="email"
+        placeholder="Email"
+        :rules="validateEmail"
+      />
+      <ErrorMessage class="flex text-red-500 mt-0.5" name="email" />
+      <label class="flex text-white mt-1 mb-1 text-lg">Password</label>
+      <Field
+        class="w-full border-solid outline-none text-base tracking-wider p-2 text-sm md:text-base lg:text-lg "
+        v-model="user.password"
+        name="password"
+        type="password"
+        placeholder="Password"
+        :rules="validatePassword"
+      />
+      <ErrorMessage class="flex text-red-500 mt-0.5" name="password" />
+
+      <div>
+        <div v-if="error" class="flex text-red-500 mt-0.5">{{ error }}</div>
+        <button
+         class="w-full h-45 bg-buttonBg mt-3 mb-3 border-0 tracking-wider p-2 md:text-base lg:text-lg"
+        >
+          LOGIN
+        </button>
+      </div>
+
+      <div class="mx-auto md:ml-2 lg:flex md:flex ">
+        <button
+          class="w-48 border-0 mb-2 mr-1 text-white bg-blue-600 p-2 text-xs md:p-0.5 lg:w-full md:w-full lg:p-1.5 2xl:p-2"
+          @click="handleFbLogin"
+        >
+          <i class="fa-brands fa-facebook mr-1 text-base"></i>Signin with
+          Facebook
+        </button>
+        <button
+          class=" w-48 border-0 mb-2 mr-1 text-white bg-blue-400 p-2 text-xs md:p-0.5 lg:w-full md:w-full lg:p-1.5 2xl:p-2"
+          @click="handleTwLogin"
+        >
+          <i class="fa-brands fa-twitter mr-1 text-base"></i>Signin with Twitter
+        </button>
+        <button
+          class=" w-48 border-0 mb-2 mr-1 text-white bg-red-700 p-2 text-xs md:p-0.5 lg:w-full md:w-full lg:p-1.5 2xl:p-2"
+          @click="handleGgLogin"
+        >
+          <i class="fa-brands fa-google mr-1 text-base"></i>Signin with Google
+        </button>
+      </div>
+    </Form>
+  </div>
+</template>
+
+<script>
+import { Form, Field, ErrorMessage } from "vee-validate";
+import { ref, reactive } from "vue";
+import { useStore } from "vuex";
+import { useRouter } from "vue-router";
+import { validationErr } from "../composables/validation.js";
+
+export default {
+  name: "login-page",
+  components: {
+    Form,
+    Field,
+    ErrorMessage,
+  },
+  setup() {
+    const user = reactive({
+      email: "",
+      password: "",
+    });
+    const error = ref(null);
+    const store = useStore();
+    const router = useRouter();
+
+    const handleLogin = async () => {
+      try {
+        // login the user with Firebase Authentication
+
+        await store.dispatch("login", {
+          email: user.email,
+          password: user.password,
+        });
+
+        router.push("/");
```

Comment:
Add success result condition for router.push @mukeshsin

Reply:
Done @meanstack9716